js/src/Ice/IdleTimeoutTransceiverDecorator.js

```diff
@@ -11,22 +11,20 @@ export class IdleTimeoutTransceiverDecorator {
         this._decoratee = decoratee;
         this._idleTimeout = idleTimeout * 1000; // Convert seconds to milliseconds
         this._timer = timer;
-        this._enableIdleCheck = enableIdleCheck;
         this._connection = connection;
+
+        // _idleCheckEnabled is initially enableIdleCheck (by default, true) unlike C++/C#/Java.
```

Comment:
The overall logic in JS is simpler/different because (i) there is no Holding state and (ii) the connection cannot be backpressured; that's why there is no enable/disable idleCheck methods in JS.